packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -34,6 +35,133 @@ typedef ScrollableWidgetBuilder = Widget Function(
   ScrollController scrollController,
 );
 
+/// Controls a [DraggableScrollableSheet].
+///
+/// Draggable scrollable controllers are typically stored as member variables in
+/// [State] objects and are reused in each [State.build]. Controllers can only
+/// be used to control one sheet at a time. A controller can be reused with a
+/// new sheet if the previous sheet has been disposed.
+///
+/// The controller's methods cannot be used until after the controller has been
+/// passed into a [DraggableScrollableSheet] and the sheet has run initState.
+class DraggableScrollableController {
+  /// Creates a draggable scrollable controller.
+  DraggableScrollableController();
+
+  _DraggableScrollableSheetScrollController? _attachedController;
+
+  /// Get the current size (as a fraction of the parent height) of the attached sheet.
+  double get size {
+    _assertAttached();
+    return _attachedController!.extent.currentSize;
+  }
+
+  /// Get the current pixel height of the attached sheet.
+  double get pixels => _attachedController!.extent.currentPixels;
```

Comment:
These getters should probably all be guarded by _assertAttached?

Reply:
Added tests cases and fixed behavior.